protostar/cheatable_starknet/callable_hint_locals/expect_events_hint_local.py

```diff
@@ -20,8 +21,14 @@ class RawExpectedEvent(TypedDict):
 
 
 class ExpectEventsHintLocal(CallableHintLocal):
-    def __init__(self, controller: ExpectEventsController):
+    def __init__(
+        self,
+        controller: ExpectEventsController,
+        test_finish_hook: Hook,
+    ):
         self._controller = controller
+        self._test_finish_hook = test_finish_hook
+        self._test_finish_hook.on(self._controller.compare_expected_and_actual_results)
```

Comment:
Wouldn't be actually better to define method `finish_hook` in `HintLocal` and presence of the HintLocal in the vm environment would trigger test runner to run this method for each local at the end? We wouldn't have to pass this convoluted `Hook` object? 🤔 
cc @Arcticae

Reply:
Ok then maybe we should just keep those checks in a separate place, and call them from test runner?